crates/sui-core/src/authority/authority_store_tables.rs

```diff
@@ -41,6 +44,12 @@ pub struct AuthorityEpochTables<S> {
     /// seems easier.
     pub(crate) pending_certificates: DBMap<TransactionDigest, TrustedCertificate>,
 
+    /// If we have obtained trusted effects (or their digests) from checkpoints (or perhaps by
+    /// querying f+1) validators, we store the effects here so that they can be used by
+    /// TransactionManager. For shared object transactions, the effects are used to set shared
+    /// locks. For all transactions, the digest is used to verify correct execution.
+    pub(crate) valid_effects: DBMap<TransactionDigest, ValidEffectsInfo>,
```

Comment:
If possible, can we consider keeping a single `pending_execution` / `pending_transaction` table combining  `pending_certificates` and `valid_effects`? e.g. `DBMap<TransactionDigest, (TrustedCertificate, Option<ValidEffectsInfo>)>`. The tuple type has appeared a few times and can be a struct type. It seems clearer for which transaction is pending, and removes any doubt that there can be a ValidEffectsInfo without corresponding TrustedCertificate with the same TransactionDigest.